### DISCUSSION TITLE
Order History

### DISCUSSION THREAD

**Discussion by jsirish** [dynamic/silverstripe-shopify#9] [Ideas]
Initial thoughts on Order History:

@muskie9 @mak001 

---

Smaller installations may not require Order History, so this could be an add-on module to a base silverstripe-shopify installation.

For larger installations, an authenticated customer should be able to view their order history. Currently Shopify does not offer a customer portal as part of their platform.

### API On Demand

Via the Shopify API, the Silverstripe website will fetch the order history for the customer, and display it on the page. Order history will not be stored in the Silverstripe database, only referenced on demand. This would require SSO implementation.

[Full field list](https://shopify.dev/docs/admin-api/rest/reference/orders/order).

### 3rd Party Service

Leverage a 3rd party Shopify add-on to provide this functionality. Presumably, the add-on would have an SSO integration with Shopify and the Silverstripe website to avoid multiple login issues. This would incur additional costs and may be an inconsistent look and feel from a UX perspective.

**Comment by mak001** [comment]
If we go the api route we would need to store the Shopify customer Id on the member object and use the customer orders endpoint.

https://shopify.dev/docs/admin-api/rest/reference/customers/customer#orders-2021-01

**Reply by jsirish** [reply]
Awesome! I have a section for Members, I'll add that next
